doc/developer/design/20230814_unbilled_replicas.md

```diff
@@ -0,0 +1,203 @@
+
+```sql
+-- Creating a managed replica set without billing
+CREATE REPLICA SET cluster.support WITH (SIZE '3xlarge', BILLED AS 'unbilled');
+-- Drop the replica set, releasing any associated resource.
+DROP REPLICA SET cluster.support;
+
+-- Alternatively, create an unmanaged replica set without billing.
+CREATE REPLICA SET cluster.support WITH (MANAGED = FALSE, BILLED AS 'unbilled');
+-- Create a replica in the replica set.
+CREATE CLUSTER REPLICA cluster.r1 SIZE '3xlarge' IN REPLICA SET support;
+-- Drop the replica set, also dropping replica `r1`.
+DROP REPLICA SET cluster.support CASCADE;
+```
+
+### Replica sets and replicas
+
+At the moment, clusters only contain replicas.
+With the introduction of replica sets, we change this property to allow different kinds of objects as part of a cluster.
+Replicas and replica sets share a common cluster namespace, so their names need to be unique within a cluster, but can be reused across clusters.
+
+At this time, users do not need to work with replica sets if they don't choose to.
+Each cluster has an implicit replica set (a.k.a, the `NULL` replica set?) that they interact with using the managed or unmanaged cluster SQL API.
+
+The following example demonstrates using the implicit replica set:
+```sql
+CREATE CLUSTER clname SIZE '3xlarge';
+```
+
+### Resources within replica sets
+
+Replicas are part of a replica sets, and their lifetime is tied to that of the replica set:
+
+```sql
+-- Cluster `cl` has an implicit replica set with one replica `r1`
+DROP REPLICA SET cl.default;
+ERROR: Replica set cl.default has objects
+HINT: Drop objects first, or specify CASCADE.
+
+DROP REPLICA SET cl.default CASCADE;
+```
+
+### Managed and unmanaged replica sets
+
+We can mix managed and unmanaged replica sets:
+
+```sql
+-- Create the a cluster with an implicit managed replica set, which include replicas `cl.r1`, `cl.r2`:
+CREATE CLUSTER cl SIZE '3xsmall', REPLICATION FACTOR 2;
+-- Create the `p1` replica set, which includes replica `cl.p1_r1`:
+CREATE REPLICA SET cl.p1 REPLICAS (p1_r1 SIZE 'large');
+```
+
+This requires a change to the names of replicas in managed clusters. Instead of
+only using the names `r1..rN`, we could use the replica set name, or `default`, as a
+prefix.
+
+The syntax to drop replicas stays the same:
+
+```sql
+-- Drop replica a replica:
+DROP CLUSTER REPLICA cl.r1;
+```
+
+### Altering replica sets
+
+* Similar to altering clusters
+  ```sql
+  ALTER REPLICA SET cl.p1 SET (REPLICATION FACTOR 0);
+  ```
+* Altering replica set name: Rename the replica set, rename replicas for managed replica sets
+  ```sql
+  ALTER REPLICA SET cl.p1 RENAME TO p2;
+  ```
+* Should we allow moving replicas between replica sets?
+  ```sql
+  ALTER CLUSTER REPLICA cl.r1 IN REPLICA SET support;
+  ERROR: Cannot move billed replica to unbilled replica set.
+  ```
+* Can we use RBAC to limit what users can do with unbilled replica sets?
+  * Owner can modify
```

Comment:
The owner in this case would be our internal user? That'd still allow customers to drop them. Seems right.